src/views/virtualmachines/list/hooks/useExistingSelectedVMs.ts

```diff
@@ -0,0 +1,27 @@
+import { useMemo } from 'react';
+
+import { V1VirtualMachine } from '@kubevirt-ui/kubevirt-api/kubevirt';
+import { getName, getNamespace } from '@kubevirt-utils/resources/shared';
+
+import { VMIdentifier } from '../selectedVMs';
+
+const useExistingSelectedVMs = (vms: V1VirtualMachine[], selectedVMs: VMIdentifier[]) => {
+  const createName = (vmId: VMIdentifier) => `${vmId.name}-${vmId.namespace}`;
+
+  const vmNames = useMemo(() => {
+    const vmsSet = new Set<string>();
```

Comment:
you can create Set directly from an array i.e. sth like `new Set(vms.map(createName))`